make check fails with gcc7

Due to the new `implicit-fallthrough` warning and `-Werror` make check fails with gcc7.
```
make  tests/pick-test
make[1]: Entering directory '/build/pick/src/pick-1.7.0'
gcc -DHAVE_CONFIG_H -I.  -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -pedantic -Wall -Werror -Wextra -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -MT tests/tests_pick_test-pick-test.o -MD -MP -MF tests/.deps/tests_pick_test-pick-test.Tpo -c -o tests/tests_pick_test-pick-test.o `test -f 'tests/pick-test.c' || echo './'`tests/pick-test.c
tests/pick-test.c: In function ‘main’:
tests/pick-test.c:76:3: error: this statement may fall through [-Werror=implicit-fallthrough=]
   child(master, slave);
   ^~~~~~~~~~~~~~~~~~~~
tests/pick-test.c:78:2: note: here
  default:
  ^~~~~~~
cc1: all warnings being treated as errors
```

--- a/compat.h
+++ b/compat.h
@@ -5,9 +5,19 @@
 #include "config.h"
 #endif
 
+#ifdef __FreeBSD__
+#define __dead __dead2
+#endif /* __FreeBSD__ */
+
+#if defined(__linux__) || defined(__CYGWIN__)
 #ifndef __dead
+#ifdef __GNUC__
+#define __dead		__attribute__((__noreturn__))
+#else
 #define __dead
 #endif
+#endif
+#endif /* __linux__ || __CYGWIN__ */
 
 #ifndef HAVE_REALLOCARRAY
 

--- a/tests/pick-test.c
+++ b/tests/pick-test.c
@@ -12,11 +12,13 @@
 #include <string.h>
 #include <unistd.h>
 
-static void	 child(int, int);
-static void	 parent(int, int, const char *);
-static char	*parsekeys(const char *);
-static void	 sighandler(int);
-static void	 usage(void);
+#include "compat.h"
+
+__dead static void	 child(int, int);
+static void		 parent(int, int, const char *);
+static char		*parsekeys(const char *);
+static void		 sighandler(int);
+__dead static void	 usage(void);
 
 static char		**pickargv;
 /*
@@ -88,7 +90,7 @@ main(int argc, char *argv[])
 	return 0;
 }
 
-static void
+__dead static void
 usage(void)
 {
 	fprintf(stderr, "usage: pick-test [-k keys] [-- argument ...]\n");
@@ -139,7 +141,7 @@ sighandler(int sig)
 	gotsig = sig == SIGCHLD;
 }
 
-static void
+__dead static void
 child(int master, int slave)
 {
 	const char	**env;

